v2/lints/etsi/lint_policy_and_qctype_are_consistent.go

```diff
@@ -0,0 +1,93 @@
+package etsi
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"encoding/asn1"
+	"fmt"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/util"
+)
+
+type consistentPolicyAndQCType struct{}
+
+func (l *consistentPolicyAndQCType) Initialize() error {
+	return nil
+}
+
+func (l *consistentPolicyAndQCType) CheckApplies(c *x509.Certificate) bool {
+	// Checks if the QcStatement id-etsi-qcs-QcCompliance is present. If yes, then this lint is triggered.
+	_, isEUQualified := util.IsQcStatementPresent(c, &util.IdEtsiQcsQcCompliance)
+	return isEUQualified
+}
+
+func policyAndTypeAreConsistent(c *x509.Certificate, policy asn1.ObjectIdentifier, policyStr string, qcType asn1.ObjectIdentifier, typeStr string) string {
+
+	// Check #1: Policy in question is present.
+	// if the EU qualified certificate does not have the policy in question, there will be no error regarding consistency.
+	if !util.HasCertPolicy(c, policy) {
+		return ""
+	}
+
+	// Check #2: Consistency of QCStatement QcType (Id-etsi-qcs-QcType) and policy.
+	_, isQcTypePresent := util.IsQcStatementPresent(c, &util.IdEtsiQcsQcType)
+
+	if isQcTypePresent && !util.HasCertEtsiQcType(c, qcType) {
+		return fmt.Sprintf("EU Qualified Certificate has policy %s but does not have the corresponding QcType '%s' (ETSI EN 319 412-4: \"Policy identifiers included in the certificate policies extension of EU Qualified Certificates shall be consistent with the EU Qualified Certificate Statements\")", policyStr, typeStr)
+	}
+
+	// Check #3: Consistency of QCStatement QcSSCD (id-etsi-qcs-QcSSCD) and policy (bidirectional).
+	_, isQscdStatementPresent := util.IsQcStatementPresent(c, &util.IdEtsiQcsQcSSCD)
+
+	qscdPolicyIsPresent := util.HasCertPolicy(c, util.IdEtsiPolicyQcpNaturalQscd) || util.HasCertPolicy(c, util.IdEtsiPolicyQcpLegalQscd)
```

Comment:
Suggestion: It may make sense to lift this to a utility function - `isQcsdPolicy` ?

It seems like you could check only based on `policy`, since you've already checked that it's present on line 41, and that would at least align with now you structured this function so that it 'could' support CA-specific OIDs in the future.

Reply:
Implemented suggestion.